compliance/collection/nodeinventorizer/cachedInventorizer.go

```diff
@@ -0,0 +1,163 @@
+package nodeinventorizer
+
+import (
+	"os"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	cmetrics "github.com/stackrox/rox/compliance/collection/metrics"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+)
+
+// InventoryScanOpts control behaviour of all node inventory related functions
+type InventoryScanOpts struct {
+	NodeName            string
+	Scanner             NodeInventorizer
+	InventoryCachePath  string
+	BackoffFilePath     string
+	BackoffWaitCallback func(time.Duration)
+}
+
+// TriggerNodeInventory creates a Node Inventory to send to Sensor.
+// To reduce strain on the Node, a linear backoff is checked before collecting an inventory.
+// Additionally, a cached inventory from an earlier invocation may be used instead of a full inventory run if it is fresh enough.
+// Note: This does not prevent strain in case of repeated pod recreation, as both mechanisms are based on an EmptyDir.
+func TriggerNodeInventory(opts *InventoryScanOpts) (*sensor.MsgFromCompliance, error) {
+	// check for existing backoff, wait for specified duration if needed, then persist the new backoff duration
+	initialBackoff := env.NodeInventoryInitialBackoff.DurationSetting()
+	currentBackoff, err := getCurrentBackoff(opts)
+	if err != nil {
+		return nil, err
+	}
+	if *currentBackoff > initialBackoff {
+		log.Warnf("Found existing backoff - last scan may have failed. Waiting %v seconds before running next inventory", currentBackoff)
```

Comment:
1. Similarly to my other logging message suggestion, this could say
```go
log.Warnf("Found existing node scan backoff marker - the last scan may have failed. Waiting %v seconds before we start the current scan", currentBackoff)
```
2. Does it actually print the value in seconds?